src/test/rpc/DepositAuthorized_test.cpp

```diff
@@ -276,11 +285,246 @@ class DepositAuthorized_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testCredentials()
+    {
+        using namespace jtx;
+
+        const char credType[] = "abcde";
+
+        Account const alice{"alice"};
+        Account const becky{"becky"};
+        Account const carol{"carol"};
+
+        Env env(*this);
+        env.fund(XRP(1000), alice, becky, carol);
+        env.close();
+
+        // carol recognize becky
+        env(credentials::create(alice, carol, credType));
+        env.close();
+        // retrieve the index of the credentials
+        auto const jv = credentials::ledgerEntry(env, alice, carol, credType);
+        std::string const credIdx = jv[jss::result][jss::index].asString();
+
+        // becky sets the DepositAuth flag in the current ledger.
+        env(fset(becky, asfDepositAuth));
+        env.close();
+
+        // becky authorize any account recognized by carol to make a payment
+        env(deposit::authCredentials(becky, {{carol, credType}}));
+        env.close();
+
+        {
+            testcase(
+                "deposit_authorized with credentials failed: empty array.");
+
+            auto args = depositAuthArgs(alice, becky, "validated");
+            args[jss::credentials] = Json::arrayValue;
+
+            auto const jv =
+                env.rpc("json", "deposit_authorized", args.toStyledString());
+            auto const& result{jv[jss::result]};
+            BEAST_EXPECT(result.isMember(jss::error));
```

Comment:
Please check the error code/message in these tests.

Reply:
added